### PR TITLE
Fix wasm and http2 uprobe rrt calculate when multi merge

### PR DESCRIPTION
**Pull request by qq906907952**
### This PR is for:
- Agent

#### Affected branches
- main
#### Checklist
- [ ] Added unit test to verify the fix.
- [ ] Verified eBPF program runs successfully on linux 4.14.x.
- [ ] Verified eBPF program runs successfully on linux 4.19.x.
- [ ] Verified eBPF program runs successfully on linux 5.2.x.

### CODE REVIEW

**Review by sharang — DISMISSED** [review]

File: agent/src/common/l7_protocol_info.rs [hunk 1]
```diff
@@ -270,6 +273,137 @@ pub trait L7ProtocolInfoInterface: Into<L7ProtocolSendLog> {
             None
         }
     }
+
+    // must have request id
+    // the main different with cal_rrt() is need to push back to lru when session not end
+    fn cal_rrt_for_multi_merge_log(&self, param: &ParseParam) -> Option<u64> {
```
Comment (sharang) [inline agent/src/common/l7_protocol_info.rs:279]: 是不是有些冗余（重复）代码？

File: agent/src/common/l7_protocol_info.rs [hunk 2]
```diff
@@ -270,6 +273,137 @@ pub trait L7ProtocolInfoInterface: Into<L7ProtocolSendLog> {
...
+
+        let mut perf_cache = param.l7_perf_cache.borrow_mut();
+        let cache_key = self.cal_cache_key(param);
+        let previous_log_info = perf_cache.rrt_cache.pop(&cache_key);
+
+        let time = param.time;
+        let msg_type: LogMessageType = param.direction.into();
+        let timeout = param.rrt_timeout as u64;
+
+        if time != 0 {
```
Comment (rvql) [inline agent/src/common/l7_protocol_info.rs:290]: 短小的分支（`time == 0`）放前面处理

File: agent/src/common/l7_protocol_info.rs (hunk 1 above)
Comment (qq906907952) [reply agent/src/common/l7_protocol_info.rs:279]: 合并了一些

File: agent/src/common/l7_protocol_info.rs [hunk 3]
```diff
@@ -270,6 +273,116 @@ pub trait L7ProtocolInfoInterface: Into<L7ProtocolSendLog> {
...
+
+        let (req_end, resp_end) = {
+            let (req_end, resp_end) = self.is_req_resp_end();
+            match param.direction {
+                PacketDirection::ClientToServer => (req_end, false),
+                PacketDirection::ServerToClient => (false, resp_end),
+            }
+        };
+
+        let Some(mut previous_log_info) = previous_log_info else {
```
Comment (rvql) [inline agent/src/common/l7_protocol_info.rs:307]: 缩进不对

Comment (qq906907952) [reply agent/src/common/l7_protocol_info.rs:307]: 改了

**Review by sharang — APPROVED** [review]

**Pull request merged**